### ISSUE TITLE
Incorrect overlapping in disabled ranges

### ISSUE THREAD

**Issue by vankop** [stylelint/stylelint#4328]
> Clearly describe the bug

During writing types in `~/lib/assignDisabledRanges` I found problem in types [here](https://github.com/stylelint/stylelint/blob/master/lib/assignDisabledRanges.js#L120) in `ranges.end` since ranges is array of disabled ranges it does not have property `end` of type array. If rewrite it to correct syntax `!!_.last(ranges).end`, getting errors in tests with incorrect overlapping in disable ranges [here](https://github.com/stylelint/stylelint/blob/master/lib/__tests__/fixtures/disableOptions/disabled-ranges-2.css), right now `block-no-empty` is closing on line 9 because of `stylelint-disable-line` inserting new disabled range for each rule and next command `stylelint-enable` throws error since all disabled ranges have `end` already

> Which rule, if any, is the bug related to?

any

> What CSS is needed to reproduce the bug?

e.g.

```css
/* stylelint-disable block-no-empty */
a {}
/* stylelint-enable */

a {} /* stylelint-disable-line color-named */
a { color: red; } /* stylelint-disable-line block-no-empty */

/* stylelint-disable block-no-empty */
a { color: red; } /* stylelint-disable-line */
/* stylelint-enable */
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
	"rules": {
			"block-no-empty": true
		},
	}
```

> Which version of stylelint are you using?

latest

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

no

> What did you expect to happen?

I think stylelint must not close `block-no-empty` disabled range on line 9

> What actually happened (e.g. what warnings or errors did you get)?

stylelint assignDisabledRanges throwing error [here](https://github.com/stylelint/stylelint/blob/master/lib/assignDisabledRanges.js#L121)

**Comment by hudochenkov** [comment]
Indeed test case looks correct and no error should be thrown.
